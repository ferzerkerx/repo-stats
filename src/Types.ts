export interface MetricsRequest {
  shouldUpdateEntries: boolean;
  config: any;
  since?: Date;
  until?: Date;
}
