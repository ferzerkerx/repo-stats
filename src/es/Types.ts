import { MetricItem } from '../metrics';

export interface ElasticSearchService {
  push(payload: MetricItem, shouldReplaceEntry?: boolean): Promise<any>;
}
