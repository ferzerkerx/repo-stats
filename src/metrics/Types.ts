export interface CollectorService<
  T extends CollectorConfig,
  K extends MetricItem
> {
  fetch(config: T): Promise<K[]>;
}

export interface MetricItem {
  id: string;
  createdAt: Date;
}

export interface CollectorConfig {}

export interface CollectorModuleFactory<
  T extends CollectorConfig,
  K extends MetricItem
> {
  collectorInstance(): CollectorService<T, K>;
  collectorConfiguration(obj: any): T;
}
