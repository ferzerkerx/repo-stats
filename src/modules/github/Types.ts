export interface GithubEntry {
  sha: string;
  createdAt: Date;
  author: string;
  filename: string;
  additions: number;
  deletions: number;
  categories?: string[];
}

export interface GithubCommitFile {
  filename: string;
  additions: number;
  deletions: number;
}

export class GithubCommit {
  sha: string;
  createdAt: Date;
  linesAdded: number;
  linesRemoved: number;
  author: string;
  message: string;
  files: GithubCommitFile[];
}

export interface GithubConfig {
  repositoryName: string;
  orgName?: string;
  since?: string;
  until?: string;
}

export interface GithubRepository {
  commits(githubConfig: GithubConfig): Promise<GithubCommit[]>;
}

export interface GithubService {
  commits(githubConfig: GithubConfig): Promise<GithubEntry[]>;
}

export interface GithubClient {
  commits(githubConfig: GithubConfig): Promise<string[]>;

  getCommitDetails(
    repositoryName: string,
    orgName: string,
    sha: string
  ): Promise<any>;
}
