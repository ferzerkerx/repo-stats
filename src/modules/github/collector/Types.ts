import { CollectorConfig, MetricItem } from '../../../metrics';

export interface GithubMetricItem extends MetricItem {
  sha: string;
  author: string;
  repositoryName: string;
  //TODO
}

export class GithubCollectorConfig implements CollectorConfig {
  repositoryName: string;
  orgName: string;
  since?: string;
  until?: string;

  constructor({ repositoryName, orgName, since = null, until = null }) {
    this.repositoryName = repositoryName;
    this.orgName = orgName;
    this.since = since;
    this.until = until;
  }
}
