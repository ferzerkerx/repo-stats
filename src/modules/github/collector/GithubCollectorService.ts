import { GithubMetricConverter } from './GithubMetricConverter';
import { CollectorService } from '../../../metrics';
import { GithubCollectorConfig, GithubMetricItem } from './Types';
import { GithubService } from '../Types';

export class GithubCollectorService
  implements CollectorService<GithubCollectorConfig, GithubMetricItem> {
  constructor(private readonly githubService: GithubService) {}

  public async fetch(
    githubCollectorConfig: GithubCollectorConfig
  ): Promise<GithubMetricItem[]> {
    const commits = await this.githubService.commits(githubCollectorConfig);
    return GithubMetricConverter.toMetricItem(commits, githubCollectorConfig);
  }
}
