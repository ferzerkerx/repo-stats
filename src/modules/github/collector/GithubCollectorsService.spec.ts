import { GithubCollectorConfig } from './Types';
import { GithubConfig, GithubEntry, GithubService } from '../Types';
import { GithubCollectorService } from './GithubCollectorService';

function testEntry(): GithubEntry {
  return {
    sha: 'someSha',
    createdAt: new Date('2018-12-06'),
    additions: 30,
    deletions: 6,
    author: 'someAuthor',
    filename: 'someFileName.txt'
  };
}

describe('GithubCollectorsService', () => {
  const githubService: GithubService = {
    commits: async (githubConfig: GithubConfig): Promise<GithubEntry[]> => {
      return [testEntry()];
    }
  };

  const githubCollectorsService: GithubCollectorService = new GithubCollectorService(
    githubService
  );

  it('should fetch githubMetrics', async () => {
    const githubCollectorConfig: GithubCollectorConfig = new GithubCollectorConfig(
      {
        repositoryName: 'someRepoName',
        orgName: 'someOrgName',
        since: '2018-11-20',
        until: '2020-11-20'
      }
    );

    const data = await githubCollectorsService.fetch(githubCollectorConfig);
    expect(data).toMatchSnapshot();
  });
});
