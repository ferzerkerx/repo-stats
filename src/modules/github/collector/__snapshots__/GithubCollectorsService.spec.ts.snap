// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubCollectorsService should fetch githubMetrics 1`] = `
Array [
  Object {
    "additions": 30,
    "author": "someAuthor",
    "createdAt": 2018-12-06T00:00:00.000Z,
    "deletions": 6,
    "filename": "someFileName.txt",
    "id": "c8a773f98009d644df4b8a3784364a61eca72f6629b897f330b67a7c5d99ed93",
    "repositoryName": "someRepoName",
    "sha": "someSha",
  },
]
`;
